worker/jwtparser/jwt.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package jwtparser
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+)
+
+// JWTParser is a parser responsible for parsing JWT tokens.
+type JWTParser struct {
```

Comment:
I think this `JWTParser` should be moved outside of the worker after seeing this all now. Specifically I would suggest something like `internal/jwt`.

The reason for this is:
- This implementation can and should survive on its own. It can be independently tested and verified as it's own component within the system.
- It also gives us a package to define the interface that `JWTParser` implements. This means that for the worker the output function can work on a well known interface type defined in internal. This allows the apiserver package and any other package to not have direct dependency on this worker anymore.

This is a convention we try to stick to for worker dependencies.

Reply:
Yeah that is fine. It isn't so much that the implementation can't live in a worker. It is that the implementation is used outside of the worker and so other component now have an import dependency onto the worker and that isn't what we want.